cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/preview/PreviewDatasetFramework.java

```diff
@@ -152,7 +152,7 @@ public void addInstance(String datasetTypeName, DatasetId datasetInstanceId,
   public void addInstance(String datasetTypeName, DatasetId datasetInstanceId,
                           DatasetProperties props,
                           @Nullable KerberosPrincipalId ownerPrincipal) throws DatasetManagementException, IOException {
-    throw new UnsupportedOperationException("Creating dataset instance with owner is not supported");
+    addInstance(datasetTypeName, datasetInstanceId, props);
```

Comment:
Should keep the exception. Just do a check if ownerPrincipal is not null throw the exception else call the addInstance which you are doing.